<a href="https://colab.research.google.com/github/cbaileyanalyst/Stroke-Insurance-Analysis/blob/main/Stroke_Health_Analysis_Cleaning_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
# Stroke/ Insurance Analysis
## Importing libraries and data sets
## Cleaning data

In [5]:
#Importing libraries
import pandas as pd
import numpy as np
import os

In [11]:
#Importing stroke dataset
df_stroke = pd.read_csv('healthcare-dataset-stroke-data.csv')

In [12]:
#Importing health insurance dataset
df_insurance = pd.read_csv('/content/states-health-insurance.csv')

In [13]:
df_stroke.head()

id  gender   age  ...   bmi   smoking_status stroke
0   9046    Male  67.0  ...  36.6  formerly smoked      1
1  51676  Female  61.0  ...   NaN     never smoked      1
2  31112    Male  80.0  ...  32.5     never smoked      1
3  60182  Female  49.0  ...  34.4           smokes      1
4   1665  Female  79.0  ...  24.0     never smoked      1

[5 rows x 12 columns]

In [14]:
df_stroke.shape

(5110, 12)

In [17]:
df_stroke.describe()

id          age  ...          bmi       stroke
count   5110.000000  5110.000000  ...  4909.000000  5110.000000
mean   36517.829354    43.226614  ...    28.893237     0.048728
std    21161.721625    22.612647  ...     7.854067     0.215320
min       67.000000     0.080000  ...    10.300000     0.000000
25%    17741.250000    25.000000  ...    23.500000     0.000000
50%    36932.000000    45.000000  ...    28.100000     0.000000
75%    54682.000000    61.000000  ...    33.100000     0.000000
max    72940.000000    82.000000  ...    97.600000     1.000000

[8 rows x 7 columns]

In [15]:
df_stroke.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


In [16]:
df_stroke.isnull().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

201 null values in the BMI column wouldn't skew the dataset too much if they were removed.

In [19]:
#Dropping the id columns because it's not relevant to my analysis
df_stroke2 = df_stroke.drop(columns= ['id'])

In [20]:
df_stroke2

gender   age  hypertension  ...   bmi   smoking_status stroke
0       Male  67.0             0  ...  36.6  formerly smoked      1
1     Female  61.0             0  ...   NaN     never smoked      1
2       Male  80.0             0  ...  32.5     never smoked      1
3     Female  49.0             0  ...  34.4           smokes      1
4     Female  79.0             1  ...  24.0     never smoked      1
...      ...   ...           ...  ...   ...              ...    ...
5105  Female  80.0             1  ...   NaN     never smoked      0
5106  Female  81.0             0  ...  40.0     never smoked      0
5107  Female  35.0             0  ...  30.6     never smoked      0
5108    Male  51.0             0  ...  25.6  formerly smoked      0
5109  Female  44.0             0  ...  26.2          Unknown      0

[5110 rows x 11 columns]

In [21]:
#Creating a subset with bmi missing values
stroke_nan = df_stroke2[df_stroke2['bmi'].isnull()==True]

In [22]:
stroke_nan

gender   age  hypertension  ...  bmi   smoking_status stroke
1     Female  61.0             0  ...  NaN     never smoked      1
8     Female  59.0             0  ...  NaN          Unknown      1
13      Male  78.0             0  ...  NaN          Unknown      1
19      Male  57.0             0  ...  NaN          Unknown      1
27      Male  58.0             0  ...  NaN          Unknown      1
...      ...   ...           ...  ...  ...              ...    ...
5039    Male  41.0             0  ...  NaN  formerly smoked      0
5048    Male  40.0             0  ...  NaN           smokes      0
5093  Female  45.0             1  ...  NaN           smokes      0
5099    Male  40.0             0  ...  NaN           smokes      0
5105  Female  80.0             1  ...  NaN     never smoked      0

[201 rows x 11 columns]

In [56]:
#Subset excluding missing values
df_stroke_clean = df_stroke2[df_stroke2['bmi'].isnull()==False]

In [57]:
#Checking for duplicate data
df_stroke_dups = df_stroke_clean[df_stroke_clean.duplicated()]

In [58]:
df_stroke_dups.shape

(0, 11)

In [59]:
df_stroke_dups

Empty DataFrame
Columns: [gender, age, hypertension, heart_disease, ever_married, work_type, Residence_type, avg_glucose_level, bmi, smoking_status, stroke]
Index: []

In [52]:
#Checking for mixed type data
for col in df_stroke_clean.columns.tolist():
  weird = (df_stroke_clean[[col]].applymap(type) != df_stroke_clean[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_stroke_clean[weird]) > 0:
    print (col)

No mixed type data in the clean stroke dataset

In [28]:
#Exporting cleaned stroke data
df_stroke_clean.to_csv('clean_stroke_Data.csv')

In [29]:
#Cleaning insurance data set

In [30]:
df_insurance.isnull().sum()

State                                           0
Uninsured Rate (2010)                           0
Uninsured Rate (2015)                           0
Uninsured Rate Change (2010-2015)               0
Health Insurance Coverage Change (2010-2015)    0
Employer Health Insurance Coverage (2015)       0
Marketplace Health Insurance Coverage (2016)    0
Marketplace Tax Credits (2016)                  0
Average Monthly Tax Credit (2016)               0
State Medicaid Expansion (2016)                 1
Medicaid Enrollment (2013)                      2
Medicaid Enrollment (2016)                      0
Medicaid Enrollment Change (2013-2016)          2
Medicare Enrollment (2016)                      0
dtype: int64

In [31]:
df_insurance.head()

State  ... Medicare Enrollment (2016)
0    Alabama   ...                     989855
1     Alaska   ...                      88966
2    Arizona   ...                    1175624
3   Arkansas   ...                     606146
4  California  ...                    5829777

[5 rows x 14 columns]

In [38]:
df_insurance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52 entries, 0 to 51
Data columns (total 14 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   State                                         52 non-null     object 
 1   Uninsured Rate (2010)                         52 non-null     object 
 2   Uninsured Rate (2015)                         52 non-null     object 
 3   Uninsured Rate Change (2010-2015)             52 non-null     object 
 4   Health Insurance Coverage Change (2010-2015)  52 non-null     int64  
 5   Employer Health Insurance Coverage (2015)     52 non-null     int64  
 6   Marketplace Health Insurance Coverage (2016)  52 non-null     int64  
 7   Marketplace Tax Credits (2016)                52 non-null     int64  
 8   Average Monthly Tax Credit (2016)             52 non-null     object 
 9   State Medicaid Expansion (2016)               51 non-null     objec

In [64]:
#Creating a subset without missing values
df_insurance2 = df_insurance.dropna()

In [65]:
df_insurance2.isnull().sum()

State                                           0
Uninsured Rate (2010)                           0
Uninsured Rate (2015)                           0
Uninsured Rate Change (2010-2015)               0
Health Insurance Coverage Change (2010-2015)    0
Employer Health Insurance Coverage (2015)       0
Marketplace Health Insurance Coverage (2016)    0
Marketplace Tax Credits (2016)                  0
Average Monthly Tax Credit (2016)               0
State Medicaid Expansion (2016)                 0
Medicaid Enrollment (2013)                      0
Medicaid Enrollment (2016)                      0
Medicaid Enrollment Change (2013-2016)          0
Medicare Enrollment (2016)                      0
dtype: int64

In [66]:
df_insurance2.head()

State  ... Medicare Enrollment (2016)
0    Alabama   ...                     989855
1     Alaska   ...                      88966
2    Arizona   ...                    1175624
3   Arkansas   ...                     606146
4  California  ...                    5829777

[5 rows x 14 columns]

In [67]:
df_dups2 = df_insurance2[df_insurance2.duplicated()]

In [68]:
df_dups2

Empty DataFrame
Columns: [State, Uninsured Rate (2010), Uninsured Rate (2015), Uninsured Rate Change (2010-2015), Health Insurance Coverage Change (2010-2015), Employer Health Insurance Coverage (2015), Marketplace Health Insurance Coverage (2016), Marketplace Tax Credits (2016), Average Monthly Tax Credit (2016), State Medicaid Expansion (2016), Medicaid Enrollment (2013), Medicaid Enrollment (2016), Medicaid Enrollment Change (2013-2016), Medicare Enrollment (2016)]
Index: []

No duplicate data in the insurance dataset

In [69]:
#Checking for mixed type data
for col in df_insurance2.columns.tolist():
  weird = (df_insurance2[[col]].applymap(type) != df_insurance2[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_insurance2[weird]) > 0:
    print (col)

No mixed type data

In [70]:
#Exporting cleaned data set
df_insurance2.to_csv('insurance_clean.csv')